backend/infrahub/database/__init__.py

```diff
@@ -91,6 +91,15 @@ def get_node_schema(self, name: str, branch: Branch | str | None = None, duplica
 
         raise ValueError("The selected node is not of type NodeSchema")
 
+    def get_generic_schema(
+        self, name: str, branch: Branch | str | None = None, duplicate: bool = True
+    ) -> GenericSchema:
+        schema = self.get(name=name, branch=branch, duplicate=duplicate)
+        if schema.is_generic_schema:
```

Comment:
Nit pick: mypy is probably disabled on this file otherwise it would raise an error as is_generic_schema does not guarantee schema type. We should probably remove these `is_generic_schema` and similar methods and use `isinstance` instead

Reply:
Yes it is, though I only copied the one we had to get a node_schema. I think we need to rewrite the getters within that class but I don't know if it should be in scope for this PR.